core/web/pipeline_runs_controller.go

```diff
@@ -131,23 +140,6 @@ func (prc *PipelineRunsController) Create(c *gin.Context) {
 		return
 	}
 
-	// only users are allowed to run jobs using int IDs - EIs not allowed
-	if isUser {
-		// Is it an int32? Then process it regardless of type
-		var jobID int32
-		jobID64, err := strconv.ParseInt(idStr, 10, 32)
-		if err == nil {
-			jobID = int32(jobID64)
-			jobRunID, err := prc.App.RunJobV2(c.Request.Context(), jobID, nil)
```

Comment:
we cannot remove this, it lets you create test runs for all job types

Reply:
for backwards compatibility (assuming I'm understanding the problem), I think we want something like this:
```
func (prc *PipelineRunsController) getJob(ctx context.Context, id string) (job.Job, error) {
	jobUUID, err := uuid.FromString(id)
	var jb job.Job
	if err == nil {
		// Look it up by UUID
		return prc.App.JobORM().FindJobByExternalJobID(jobUUID)
	}
	// Look it up by int32
	jobID, err := strconv.ParseInt(id, 10, 32)
	if err != nil {
		return jb, err
	}
	return prc.App.JobORM().FindJob(ctx, int32(jobID))
}

// Create triggers a pipeline run for a job.
// Example:
// "POST <application>/jobs/:ID/runs"
func (prc *PipelineRunsController) Create(c *gin.Context) {
	respondWithPipelineRun := func(jobRunID int64) {
		pipelineRun, err := prc.App.PipelineORM().FindRun(jobRunID)
		if err != nil {
			jsonAPIError(c, http.StatusInternalServerError, err)
			return
		}
		res := presenters.NewPipelineRunResource(pipelineRun, prc.App.GetLogger())
		jsonAPIResponse(c, res, "pipelineRun")
	}

	bodyBytes, err := ioutil.ReadAll(c.Request.Body)
	if err != nil {
		jsonAPIError(c, http.StatusUnprocessableEntity, err)
		return
	}
	idStr := c.Param("ID")

	user, isUser := auth.GetAuthenticatedUser(c)
	ei, _ := auth.GetAuthenticatedExternalInitiator(c)
	authorizer := webhook.NewAuthorizer(prc.App.GetSqlxDB().DB, user, ei)

	jb, err := prc.getJob(c.Request.Context(), idStr)
	if err != nil {
		jsonAPIError(c, http.StatusUnprocessableEntity, err)
		return
	}
	switch jb.Type {
	case job.Webhook:
		canRun, err2 := authorizer.CanRun(c.Request.Context(), prc.App.GetConfig(), jb.ExternalJobID)
		if err2 != nil {
			jsonAPIError(c, http.StatusInternalServerError, err2)
			return
		}
		if canRun {
			jobRunID, err3 := prc.App.RunWebhookJobV2(c.Request.Context(), jb.ExternalJobID, string(bodyBytes), pipeline.JSONSerializable{})
			if errors.Is(err3, webhook.ErrJobNotExists) {
				jsonAPIError(c, http.StatusNotFound, err3)
				return
			} else if err3 != nil {
				jsonAPIError(c, http.StatusInternalServerError, err3)
				return
			}
			respondWithPipelineRun(jobRunID)
		} else {
			jsonAPIError(c, http.StatusUnauthorized, errors.Errorf("external initiator %s is not allowed to run job %s", ei.Name, jobUUID))
		}
		return
	default:
		if isUser {
			jobRunID, err := prc.App.RunJobV2(c.Request.Context(), jb.ID, nil)
			if err != nil {
				jsonAPIError(c, http.StatusInternalServerError, err)
				return
			}
			respondWithPipelineRun(jobRunID)
		}
		return
	}
	jsonAPIError(c, http.StatusUnprocessableEntity, errors.New("bad job ID"))
}

```

Reply:
Thanks.  Let me see if this works.

Also is there any way of modifying the UI to send the UUID instead of the job ID.  That might be cleaner.

Reply:
@jkongie thoughts on modifying the UI here?